src/typechecker/FStar.TypeChecker.Rel.fs

```diff
@@ -3089,7 +3091,10 @@ let solve_and_commit env probs err =
   if Env.debug env <| Options.Other "RelBench" then
     BU.print1 "solving problems %s {\n"
       (FStar.Common.string_of_list (fun p -> string_of_int (p_pid p)) probs.attempting);
-  let (sol, ms) = BU.record_time (fun () -> solve env probs) in
+  let (sol, ms)  = 
+    P.profile (fun () -> solve env probs)
+      (fun() -> FStar.Common.string_of_list (fun p -> string_of_int (p_pid p)) probs.attempting)
+      Options.ProfileSMT in
```

Comment:
This is actually counting the time spend in the unification engine, not the SMT solver.

It's a useful thing to measure too, but should be put in a different bucket. (e.g., Unification time)

Reply:
I should've clued in sooner by the file name, guess I got too exciteded when I saw "solve". I changed it now so that it is profiling z3_job, the same way as the elapsed time for --query_stats